### ISSUE TITLE
Password Reset issue

### ISSUE THREAD

**Issue by toconnell** [toconnell/kdm-manager#535]
From the blog:

> Anomandras  November 4, 2019 at 5:26 AM
> Hello, my (non-subscription) players have to password reset once per week. It's not that they forget their pw once a week, it just wont be accepted anymore.
> We got accustomed to that, but sadly, pw resets also dont work anymore. We tried to use new accounts but since we don't play every week we would have to create new accounts for every session.
> I just tried to reset a password myself with a brandnew account, but i only get a blank message (black page, red bar, no text) but no email. According to my players, that's the status for more than 6 weeks. Please help :-)

**Comment by toconnell** [comment]
Ok, first off, this appears NOT to be related to _changing_ or updating your password in the webapp.

Rather, it has to do with trying to reset it from the login screen. I just got this API error attempting that in production:

```
==> kdm-manager-api/logs/error.log <==
[2019-11-04 10:06:50] ERROR:    module 'app.utils' has no attribute 'get_application_url'
Traceback (most recent call last):
  File "/home/toconnell/kdm-manager-api/venv/lib/python3.6/site-packages/flask/app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/toconnell/kdm-manager-api/venv/lib/python3.6/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/toconnell/kdm-manager-api/app/utils/crossdomain.py", line 56, in wrapped_function
    resp = flask.make_response(func(*args, **kwargs))
  File "/home/toconnell/kdm-manager-api/app/routes.py", line 341, in reset_password
    return users.initiate_password_reset()
  File "/home/toconnell/kdm-manager-api/app/models/users.py", line 200, in initiate_password_reset
    netloc = utils.get_application_url()
AttributeError: module 'app.utils' has no attribute 'get_application_url'
```

**Comment by toconnell** [comment]
Yeah, OK: looks like we missed a couple of critical bits from the _utils.py_ module in the 1.0.0 release. I'm going to be patching those back in and making some improvements to automatic failure email handling.

**Comment by toconnell** [comment]
Alright, system-side issues appear to be addressed, but emails still aren't going out. Researching...

**Comment by toconnell** [comment]
This is fixed in production. I found a few problems in the legacy app and the API, however, that will need addressing, so I'm going to keep this one open to remind me that I need to handle this biz.

**Comment by toconnell** [comment]
Authentication issues are addressed in 3.105.1888.
